src/test/java/chatbot/ParserTest.java

```diff
@@ -0,0 +1,148 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+
+import chatbot.exceptions.DeleteMissingFieldException;
+import chatbot.exceptions.EventMissingFieldException;
+import chatbot.exceptions.InvalidTaskIndexException;
+import chatbot.exceptions.MarkMissingFieldException;
+import chatbot.exceptions.MissingFieldException;
+import chatbot.tasks.EventTask;
+import chatbot.tasks.Task;
+
+public class ParserTest {
+    @Test
+    public void markTest1() {
+        String[] commandWords = new String[]{"mark"};
+        try {
+            Parser.parseMarkCommand(commandWords);
+            fail();
+        } catch (MissingFieldException e) {
+            assertTrue(e instanceof MarkMissingFieldException);
+        } catch (InvalidTaskIndexException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void markTest2() {
+        String[] commandWords = new String[]{"mark", "2103"};
+        try {
+            assertEquals(Parser.parseMarkCommand(commandWords), 2103);
+        } catch (MissingFieldException e) {
+            fail();
+        } catch (InvalidTaskIndexException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void markTest3() {
+        String[] commandWords = new String[]{"mark", "a1"};
+        try {
+            Parser.parseMarkCommand(commandWords);
+            fail();
+        } catch (MissingFieldException e) {
+            fail();
+        } catch (InvalidTaskIndexException e) {
+            assertTrue(true);
+        }
+    }
+
+    @Test
+    public void deleteTest1() {
+        String[] commandWords = new String[]{"delete"};
+        try {
+            Parser.parseDeleteCommand(commandWords);
+            fail();
+        } catch (MissingFieldException e) {
+            assertTrue(e instanceof DeleteMissingFieldException);
+        } catch (InvalidTaskIndexException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void deleteTest2() {
+        String[] commandWords = new String[]{"delete", "2103"};
+        try {
+            assertEquals(Parser.parseDeleteCommand(commandWords), 2103);
+        } catch (MissingFieldException e) {
+            fail();
+        } catch (InvalidTaskIndexException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void deleteTest3() {
+        String[] commandWords = new String[]{"delete", "a1"};
+        try {
+            Parser.parseDeleteCommand(commandWords);
+            fail();
+        } catch (MissingFieldException e) {
+            fail();
+        } catch (InvalidTaskIndexException e) {
+            assertTrue(true);
+        }
+    }
+
+    @Test
+    public void eventTest1() {
+        String command = "event /from 1 /to 2";
+        try {
+            Parser.parseEventTaskCommand(command);
+            fail();
+        } catch (MissingFieldException e) {
+            assertTrue(e instanceof EventMissingFieldException);
```

Comment:
`Assertions.assertThrows` can be a good alternative to this that would result in a more concise test.

Reply:
Yep it would be a better way when I add testing self-defined Exceptions in the future. Thanks for the suggestion!